Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -352,28 +353,17 @@ int CLuaResourceDefs::GetResourceGUIElement(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM)
+CClientEntity* CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM, std::optional<CResource*> resource)
 {
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
+    CResource* pResource = resource.value_or(&lua_getownerresource(luaVM));
 
-    if (!argStream.HasErrors())
+    CClientEntity* resourceDynamicRoot = pResource->GetResourceDynamicEntity();
```

Comment:
The server's counterpart of this function haven't this check. Maybe make them identical?

Reply:
Done